pkg/daemon/api.go

```diff
@@ -0,0 +1,80 @@
+package daemon
+
+import (
+	"io"
+	"os"
+	"time"
+
+	"github.com/cybozu-go/necoperf/pkg/rpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	maxTimeout = 10 * time.Minute // 10 minute
+)
+
+func (d *DaemonServer) Profile(req *rpc.PerfProfileRequest, stream rpc.NecoPerf_ProfileServer) error {
```

Comment:
Is this method thread-safe?
What happens if multiple clients call `Profile` for the same process?

Reply:
You have some options.

- Rename the name of output file. This method could be executed concurrently.
- Use semaphore. You can limit the number of concurrent executions.